Vagrantfile

```diff
@@ -9,14 +9,32 @@ if not File.file?(config_file)
 end
 pubstack_config = YAML::load_file(config_file)
 
-# Validate sites' docroots:
-if pubstack_config.include? 'sites'
+# If the synced folder option is specified, fix the documentroot option
+# for every site.
+if pubstack_config.include? 'synced_folder'
+  # It doesn't make sense to specify both synced_folder and shared_folders,
+  # and the path to the documentroot in the VM will break, so just bail
+  # if both are specified for whatever reason.
+  if pubstack_config.include? 'shared_folders'
+    raise Vagrant::Errors::VagrantError.new, 'You may only use the synced_folder option if shared_folders is not specified. Please fix your config.yml and try again.'
+  end
+
   pubstack_config['sites'].each {|site|
```

Comment:
I think we should still check `if pubstack_config.include? 'sites'`. Not sure why anyone wouldn't have sites defined in their config, but it _could_happen... Also, if `shared_folders` is defined in config, then we aren't checking that the sites' documentroot directories exist on the host machine -- I guess that's not necessary because that person would then "know what they are doing"?  :smile: